### PR TITLE
Optimize setup.py

### PR DESCRIPTION
**Pull request by ademhilmibozkurt**
Pull request for issue #260. Homebrew was seperated and called from setup.py

### CODE REVIEW

File: homebrew.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+from re import search
+from sys import platform
+from subprocess import check_output, CalledProcessError
+
+class HomeBrew:
+    def __init__(self, libdirs: list[str], incdirs: list[str]):
```
Comment (carlkidcrypto) [inline homebrew.py:6]: make sure to run Black Formatter too.

**Review by carlkidcrypto — CHANGES_REQUESTED** [review]

File: homebrew.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+from re import search
+from sys import platform
+from subprocess import check_output, CalledProcessError
+
+class HomeBrew:
```
[the remark this answers is not in the document]
Comment (ademhilmibozkurt) [reply homebrew.py:6]: I added properties but can't understand actually what you ask for. Please, review new commit and make contact.

[the remark this answers is not in the document]
Comment (carlkidcrypto) [reply homebrew.py:6]: Perhaps, it is easier to say what I want in the setup.py.

```python3

other code here
...


hb = HomeBrew()

# Make HomeBrew() getters return None or "" if libdirs or incdirs aren't found.
if hb.libdirs and hb.incdirs:
     libdirs = hb.libdirs
     incdirs = hb.incdirs
     homebrew_version = hb.homebrew_version
    homebrew_netsnmp_version = hb.homebrew_netsnmp_version
    homebrew_openssl_version = hb.homebrew_openssl_version

print(f"in_tree: {in_tree}")
print(f"compile_args: {compile_args}")
print(f"link_args: {link_args}")
print(f"platform: {platform}")
print(f"system_netsnmp_version: {str(system_netsnmp_version).strip()}")
print(f"homebrew_version: {str(homebrew_version).strip()}")
print(f"homebrew_netsnmp_version: {homebrew_netsnmp_version}")
print(f"homebrew_openssl_version: {homebrew_openssl_version}")
print(f"macports_version: {str(macports_version).strip()}")
print(f"macports_netsnmp_version: {macports_netsnmp_version}")
print(f"macports_openssl_version: {macports_openssl_version}")
print(f"libs: {libs}")
print(f"libdirs: {libdirs}")
print(f"incdirs: {incdirs}")

....

rest of code here
```

**Review by carlkidcrypto — CHANGES_REQUESTED** [review]
merge up from main.

**Review by carlkidcrypto — CHANGES_REQUESTED** [review]
Merge up from main.

File: homebrew.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
+from re import search
+from sys import platform
+from subprocess import check_output, CalledProcessError
+
+
+class HomeBrew:
+    def __init__(self):
+        self._libdirs = self.libdirs
```
Comment (carlkidcrypto) [inline homebrew.py:8]: These should now be initiated to `None`

File: homebrew.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        self._incdirs = self.incdirs
+        self._check_brew_isinstalled()
```
Comment (carlkidcrypto) [inline homebrew.py:10]: Don't forget to also call the other helpers here to:

`    def _check_netsnmp_isinstalled(self) -> list[str]:`

`    def _check_openssl_isinstalled(self) -> list[str]:`

**Review by carlkidcrypto — CHANGES_REQUESTED** [review]

File: homebrew.py (hunk 3 above)
Comment (ademhilmibozkurt) [reply homebrew.py:8]: Done

File: homebrew.py (hunk 4 above)
Comment (ademhilmibozkurt) [reply homebrew.py:10]: These two operations calling from related functions.

**Review by carlkidcrypto — CHANGES_REQUESTED** [review]
Merge up from main and resolve conflicts.

**Comment by ademhilmibozkurt** [comment]
Okay. working on it.

**Comment by ademhilmibozkurt** [comment]
Will merge from main.

**Pull request closed without merging**